monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ComponentControllerV1.java

```diff
@@ -498,11 +504,40 @@ private static void setComponentAnnotations(ComponentDescriptor<?> componentDesc
 
     static void setPropertyType(ComponentDescriptor<?> descriptor, ConfiguredPropertyDescriptor propertyDescriptor,
             ComponentList.PropertyInfo propInfo) {
-        // TODO: avoid instanceof by extending the basic ComponentDescriptor
-        // interface (maybe add getter for property "Type" in addition to
-        // "Class" ? )
 
-        SchemaFactoryWrapper visitor = new SchemaFactoryWrapper();
+        // Special support for enumerations in case we are publishing info about a remote transformer from another server.
+        // (Client -> Server1 -> Server2)
+        // In this case if Server1 doesn't have the enumeration class on its classpath, the json schema cannot
+        // cannot be generated from a class => We do it "by hand" here.
+        if(EnumerationValue.class.isAssignableFrom(propertyDescriptor.getBaseType())
+                && propertyDescriptor instanceof EnumerationProvider) {
+            EnumerationValue[] values = ((EnumerationProvider) propertyDescriptor).values();
```

Comment:
As well as most of this method (I know, I'm being a big anal about it... I blame @kaspersorensen :D)